app/src/main/java/fr/free/nrw/commons/filepicker/PickedFiles.java

```diff
@@ -64,10 +64,11 @@ private static void writeToFile(InputStream in, File file) {
             while ((len = in.read(buf)) > 0) {
                 out.write(buf, 0, len);
             }
-            out.close();
-            in.close();
         } catch (Exception e) {
             e.printStackTrace();
+        } finally {
+            out.close();
+            in.close();
```

Comment:
I believe this code won't work as-is. Few points:

1. `out` isn't accessible in the finally block given where it is declared right now.
2. The `close` method throws an exception which we could need to handle accordingly.

I suppose using try-with-resources block would simplify the code here 🤔

Reply:
Is there any reason why this method should close the `InputStream` stream supplied by the caller?

Reply:
something along these lines came to mind:

```java
private static void writeToFile(File src, File dst) throws IOException {
        try (InputStream in = new FileInputStream(src);
               OutputStream out = new FileOutputStream(dst)) {
                byte[] buf = new byte[1024];
                int len;
                while ((len = in.read(buf)) > 0) {
                    out.write(buf, 0, len);
                }
       } catch (..) {
           ...
       }
}
```